### PR TITLE
[xbyak] Add new port

### LINKED ISSUE #11673: [New Port Request] Xbyak
Library name: 
Xbyak

Library description:
Xbyak is a JIT assembler for x86(IA-32)/x64(AMD64, x86-64).

Source repository URL:
https://github.com/herumi/xbyak

### PR DESCRIPTION
**Pull request by Ryan-rsm-McKenzie**
**Describe the pull request**

Adds [xbyak](https://github.com/herumi/xbyak) to vcpkg

- What does your PR fix? Fixes #11673

- Which triplets are supported/not supported? Have you updated the CI baseline?
Header only, however only supports x86/x64 Windows/Linux/OSX

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/xbyak/CONTROL [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Source: xbyak
+Version: 5.911
```
Comment (LilyWangL) [inline ports/xbyak/CONTROL:2]: Can you please add the Homepage for this port in the CONTROL file?

File: ports/xbyak/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#header-only library
+include(vcpkg_common_functions)
```
Comment (LilyWangL) [inline ports/xbyak/portfile.cmake:2]: `include(vcpkg_common_functions)` is deprecated. Please remove it.

File: ports/xbyak/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO herumi/xbyak
+    REF v5.911
+    SHA512 3cb6b1e73d7dc2a23dcb31310720f0d4b83b62cfc69ff09eeefafe032e58e04574419f668d5ca957e8fc21e679e25da059f6e93724949e0ff1fcaa6779b88bdd
+    HEAD_REF master
+)
+
+# handle license file
+file(INSTALL ${SOURCE_PATH}/COPYRIGHT DESTINATION ${CURRENT_PACKAGES_DIR}/share/xbyak RENAME copyright)
```
Comment (LilyWangL) [inline ports/xbyak/portfile.cmake:13]: Can you please use command `file(INSTALL ${SOURCE_PATH}/COPYRIGHT DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` to handle copyright?

**Review by LilyWangL — CHANGES_REQUESTED** [review]

**Comment by dan-shaw** [comment]
Thanks for the PR!

**Pull request merged**